### PR TITLE
Automate release version generation [main] 

### PR DESCRIPTION
**Pull request by gururajsh**
While releasing CLI, take the input from the user for minor/patch bump. Based on the selection, calculate the next release version. New build version need not be stored in a file going forward.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -3,7 +3,6 @@ NODES ?= 10
 PACKAGES ?= api actor command types util version integration/helpers
 LC_ALL = "en_US.UTF-8"
 
-CF_BUILD_VERSION ?= $$(cat BUILD_VERSION) # TODO: version specific
```
Comment (a-b) [inline Makefile:6]: same https://github.com/cloudfoundry/cli/pull/3268#discussion_r1819183154

**Review by a-b — APPROVED** [review]

File: bin/bump-version
Comment (a-b) [inline bin/bump-version:1]: prise: so happy we are cleaning up this

File: bin/generate-release-notes
Comment (a-b) [inline bin/generate-release-notes:1]: praise: happy we are getting rid of legacy craft

File: Makefile (hunk 1 above)
Comment (gururajsh) [reply Makefile:6]: Fixed

**Review by joaopapereira — APPROVED** [review]
LGTM

**Pull request merged**